addons/mail/static/src/core/common/thread_model.js

```diff
@@ -287,11 +287,11 @@ export class Thread extends Record {
     pid;
 
     get accessRestrictedToGroupText() {
-        if (!this.authorizedGroupFullName) {
+        if (!this.group_public_id) {
```

Comment:
```suggestion
        if (!this.group_public_id?.full_name) {
```
We might want to handle the case of name not received too.